packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -97,52 +116,113 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
 
   /// The locale for which the values of this class's localized resources
   /// have been translated.
-  final Locale locale;
+  Locale locale;
+
+  String get _localeName {
+    final String localeName = locale.countryCode.isEmpty ? locale.languageCode : locale.toString();
+    return Intl.canonicalizedLocale(localeName);
+  }
+
+  // TODO(hmuller): the rules for mapping from an integer value to
+  // "one" or "two" etc. are locale specific and an additional "few" category
+  // is needed. See http://cldr.unicode.org/index/cldr-spec/plural-rules
+  String _nameToPluralValue(int count, String key) {
+    String text;
+    if (count == 0)
+      text = _nameToValue['${key}Zero'];
+    else if (count == 1)
+      text = _nameToValue['${key}One'];
+    else if (count == 2)
+      text = _nameToValue['${key}Two'];
+    else if (count > 2)
+      text = _nameToValue['${key}Many'];
+    text ??= _nameToValue['${key}Other'];
+    assert(text != null);
+    return text;
+  }
+
+  NumberFormat get _integerFormat {
+    String localeName = _localeName;
+    if (!NumberFormat.localeExists(localeName))
+      localeName = 'en';
```

Comment:
I would default to `int.toString()` in that case. Defaulting to English is a bit dangerous. Consider the case where NumberFormat doesn't know about a French-based locale. It'll turn "12345" into "12,345" instead of "12.345". The user will be very confused because that's a factor of 1000 off from what they will think it is.